lstchain/visualization/plot_dl2.py

```diff
@@ -132,44 +132,45 @@ def plot_features(data, truehadroness=False):
     plt.xlabel(r"Time gradient")
 
 
-def plot_e(data,truehadroness=False):
-    
-    """Plot the performance of reconstructed Energy. 
+def plot_e(data, Nbins, emin, emax, truehadroness=False):
+
+    """Plot the performance of reconstructed Energy.
 
     Parameters:
     -----------
     data: pandas DataFrame
-    
+
     truehadroness:
-    True: True gammas and proton events are plotted (they are separated using true hadroness). 
+    True: True gammas and proton events are plotted
+    (they are separated using true hadroness).
     False: Gammas and protons are separated using reconstructed hadroness (hadro_rec)
-    
+
     """
     hadro = "hadro_rec"
     if truehadroness:
         hadro = "hadroness"
-    
-    gammas = data[data[hadro]==0] 
-    
+
+    gammas = data[data[hadro]==0]
+
     plt.subplot(221)
-    difE = ((gammas['mc_energy']-gammas['e_rec'])*np.log(10))
-    section = difE[abs(difE) < 1.5]
-    mu, sigma = norm.fit(section)
-    print(mu, sigma)
-    n, bins, patches = plt.hist(difE,100,
-                                density=1,alpha=0.75)
-    y = norm.pdf( bins, mu, sigma)
-    plt.plot(bins, y, 'r--', linewidth=2)
-    plt.xlabel('$(log_{10}(E_{gammas})-log_{10}(E_{rec}))*log_{N}(10)$',
-               fontsize=10)
-    plt.figtext(0.15,0.7,'Mean: '+str(round(mu,4)),
-                fontsize=10)
-    plt.figtext(0.15,0.65,'Std: '+str(round(sigma,4)),
-                fontsize=10)
-    
+
+    difE = np.log(10**data['e_rec']/10**data['mc_energy'])
+    means_result = scipy.stats.binned_statistic(
+        data['mc_energy'],[difE,difE**2],
+        bins=Nbins,range=(emin, emax),statistic='mean')
+    means, means2 = means_result.statistic
```

Comment:
`means_result.statistic` should return a single array...?

Reply:
I'm asking the binned_statistics function to return the mean of difE and difE**2, so it returns two arrays, mean (mean of difE) and mean2 (mean of difE**2) 
_scipy.stats.binned_statistic( data['mc_energy'], here->[difE,difE**2]<--, ..., )_